tests/integration_tests.rs

```diff
@@ -670,3 +674,38 @@ fn do_not_panic_regression_tests() {
             .success();
     }
 }
+
+#[test]
+fn do_not_detect_different_syntax_for_stdin_and_files() {
+    let file = "regression_tests/issue_985.js";
+
+    let output_for_file = bat()
+        .arg("--color=always")
+        .arg("--map-syntax=*.js:Markdown")
+        .arg(&format!("--file-name={}", file))
+        .arg("--style=plain")
+        .arg(file)
+        .assert()
+        .success()
+        .get_output()
+        .stdout
+        .clone();
```

Comment:
not that it matters too much here, but you can avoid the `.clone` by using:
``` rust
    let cmd_for_file = bat()
        .arg("--color=always")
        .arg("--map-syntax=*.js:Markdown")
        .arg(&format!("--file-name={}", file))
        .arg("--style=plain")
        .arg(file)
        .assert()
        .success();
    let output_for_file = &cmd_for_file.get_output().stdout;
```
similar below.

Reply:
Great suggestion! Fixed